### ISSUE TITLE
训练时发现输入进compute_nlg_metrics的preds被截断

### ISSUE THREAD

**Issue by Yiiiiie** [modelscope/ms-swift#5963]
**Describe the bug**
训练时想使用custom metric，发现全是format_error，调试时发现输入进compute_nlg_metrics的preds被截断

运行脚本如
swift sft \
    --model xxxxxxxxxxxxxxx \
    --model_type qwen2_5_vl \
    --train_type lora \
    --dataset 'xxxxxxxxxxxxxx'\
    --split_dataset_ratio 0.05 \
    --torch_dtype bfloat16 \
    --attn_impl flash_attn \
    --freeze_vit true \
    --freeze_llm false \
    --freeze_aligner true \
    --predict_with_generate True \
    --num_train_epochs 5 \
    --save_steps 200 \
    --per_device_train_batch_size 1 \
    --per_device_eval_batch_size 1 \
    --learning_rate 1e-4 \
    --lora_rank 8 \
    --lora_alpha 32 \
    --gradient_accumulation_steps 4 \
    --eval_steps 1 \
    --logging_steps 1 \
    --max_length 25600 \
    --output_dir xxxxxxxxxxxxxxxxxxxxxx \
    --dataloader_num_workers 4 \
    --dataset_num_proc 8 \
    --report_to tensorboard \

metric.py

def compute_nlg_metrics(prediction) -> Dict[str, float]:
    preds, labels = prediction[0], prediction[1]
    new_preds, new_labels = [], []
    for i in range(preds.shape[0]):
        new_preds.append(Serializer.from_tensor(preds[i]))
        new_labels.append(Serializer.from_tensor(labels[i]))
   ##########################################
    这里发现new_preds只有一半
    new_labels是完整的回答
   ##########################################
    return compute_rouge_bleu(new_preds, new_labels)

**Your hardware and system info**
CUDA Version: 12.6
NVIDIA A100
ms-swift main
transformers 4.55.0
torch 2.6.0
